### PR TITLE
Testcase: Multiple clones of PGSQL Workloads PVCs

### PR DESCRIPTION
**Pull request by Akarsha-rai**
Signed-off-by: Akarsha <akrai@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/pgsql.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 from ocs_ci.ocs.resources.ocs import OCS
 from ocs_ci.ocs import constants
 from subprocess import CalledProcessError
-from ocs_ci.ocs.resources.pod import get_all_pods, get_pod_obj, get_operator_pods
+from ocs_ci.ocs.resources.pod import get_all_pods, get_pod_obj, get_operator_pods, get_file_path
```
Comment (PrasadDesala) [inline ocs_ci/ocs/pgsql.py:17]: Please break the line, it has exceeded 79 char

File: ocs_ci/ocs/pgsql.py [hunk 2]
```diff
@@ -132,25 +138,27 @@ def create_pgbench_benchmark(
...
-            try:
-                if len(pgbench_pods) == replicas:
+
+        if wait:
+            # Confirm that expected pgbench pods are spinned
+            log.info("Searching the pgbench pods by its name pattern")
+            timeout = timeout if timeout else 300
+            for pgbench_pods in TimeoutSampler(
+                timeout, replicas, get_pod_name_by_pattern,
+                'pgbench-1-dbs-client', RIPSAW_NAMESPACE
```
Comment (PrasadDesala) [inline ocs_ci/ocs/pgsql.py:148]: Does the patten "pgbench-1-dbs-client" remain same even after pgbench-benchmark name is provided?

File: tests/e2e/lifecycle/pvc_clone/test_pgsql_pvc_clone.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                postgres_name=f'postgres-cloned-{i}', wait=False
+            )
+
+            # Run benchmark on parent PVC
+            pgsql.create_pgbench_benchmark(replicas=3, pgbench_name=f'pgbench-{i}', wait=False)
+
+            # Wait till pgbench client pods up
+            wait_time = 180
+            log.info(f"Waiting {wait_time} seconds for pgbench client pods to be up")
+            time.sleep(180)
```
Comment (PrasadDesala) [inline tests/e2e/lifecycle/pvc_clone/test_pgsql_pvc_clone.py:109]: Can we use wait_for_pgbench_status?

File: ocs_ci/ocs/pgsql.py [hunk 4]
```diff
@@ -536,3 +544,24 @@ def attach_pgsql_pod_to_claim_pvc(
...
+
+        """
+        # Get the used file space on a mount point
+        for pod_obj in pod_obj_list:
+            filepath = get_file_path(pod_obj, 'pgdata')
+            filespace = pod_obj.exec_cmd_on_pod(
+                command=f"du -sh {filepath}", out_yaml_format=False
+            )
+            filespace = filespace.split()[0]
+            pod_obj.filespace = filespace
```
Comment (PrasadDesala) [inline ocs_ci/ocs/pgsql.py:566]: We have an function get_used_space_on_mount_point in ocs_ci/ocs/resources/pod.py. 
Can this be replaced with that?

Comment (PrasadDesala) [reply ocs_ci/ocs/pgsql.py:566]: But I see that function is returning used space in percentage though

Comment (Akarsha-rai) [reply ocs_ci/ocs/pgsql.py:566]: Used space in mount point won't be much difference after each iteration , so here we are not checking for the used space instead checking for the file size that returns in mb.

File: tests/e2e/lifecycle/pvc_clone/test_pgsql_pvc_clone.py (hunk 3 above)
Comment (Akarsha-rai) [reply tests/e2e/lifecycle/pvc_clone/test_pgsql_pvc_clone.py:109]: The problem here is pgbench client pod takes some time to come up. So if we use `wait_for_pgbench_status` immediately it checks for existing pgbench-client pods only.

File: ocs_ci/ocs/pgsql.py (hunk 2 above)
Comment (Akarsha-rai) [reply ocs_ci/ocs/pgsql.py:148]: no changes for `pgbench-1-dbs-client`,  pattern will remain same

File: ocs_ci/ocs/pgsql.py (hunk 1 above)
Comment (Akarsha-rai) [reply ocs_ci/ocs/pgsql.py]: done

File: ocs_ci/ocs/pgsql.py (hunk 4 above)
Comment (PrasadDesala) [reply ocs_ci/ocs/pgsql.py:571]: ack

**Comment by Akarsha-rai** [comment]
Locally verified

[test_pgsql_pvc_clone.txt](https://github.com/red-hat-storage/ocs-ci/files/5493251/test_pgsql_pvc_clone.txt)

**Review by jilju — APPROVED** [review]
LGTM

**Review by PrasadDesala — APPROVED** [review]

**Review by vavuthu — APPROVED** [review]

**Pull request merged**